### DISCUSSION TITLE
How to set TAURI_PRIVATE_KEY

### DISCUSSION THREAD

**Discussion by DrAugus** [tauri-apps/tauri#4451] [Q&A] [answered]
Error running CLI: A public key has been found, but no private key. Make sure to set `TAURI_PRIVATE_KEY` environment variable.

**Comment by FabianLars** [comment, marked as answer]
<https://tauri.app/v1/guides/distribution/updater#signing-updates>

When you generated your keys you should've received _2_ keys, a public one and a private one. You need to set a `TAURI_PRIVATE_KEY` _environment variable_ with the contents of the private key.

**Reply by johnsonkw** [reply]
I have a .env file but with TAURI_PRIVATE_KEY="key is here" in the root of my project but it is not working... Any ideas? 
Im getting the same error mentioned above

**Reply by FabianLars** [reply]
.env files are not picked up. We're talking about the actual system/terminal environment.

**Reply by boan-anbo** [reply]
> 

For example, if you use Windows, check out https://phoenixnap.com/kb/windows-set-environment-variable

If you use mac, https://phoenixnap.com/kb/set-environment-variable-mac

Also, on Mac, if you are building Tauri using `sudo`, you will need to use `sudo -E` in order to expose environment variables.

Note the environment variable can be terminal dependent, e.g. CMD & Powershell might not share certain environment variables. So make sure the terminal where you run `Tauri build` has this environment variable. You can check with commands like `echo $ENV` to see if the variable is already set.

**Reply by guido-visser** [reply]
For anyone reading this, if you have added the keys (both `TAURI_PRIVATE_KEY` and `TAURI_KEY_PASSWORD`) to your environment and it keeps saying they're not there, restart your PC. That worked for me :)
